neuro_san/coded_tools/rag.py

```diff
@@ -36,26 +36,6 @@ class Rag(CodedTool):
     and RAG is applied afterward to synthesize a meaningful answer from the linked content.
     """
 
-    def invoke(self, args: Dict[str, Any], sly_data: Dict[str, Any]):
```

Comment:
No need to provide an invoke() method that does nothing.
That's what the CodedTool base implementation is for.